src/formula.ts

```diff
@@ -137,7 +123,7 @@ export default class Formula extends Node {
    * @param {Statement} statement - An existing constructed statement to add
    */
   addStatement (statement: Quad): number {
-    return this.statements.push(statement)
+    return this.add(statement) as number
```

Comment:
The `as number` is weird. If the output type of `add` depends on the input parameter type, then we might want to issue multiple signatures for `add` (which is possible in TypeScript).